src/include/txn.i

```diff
@@ -68,8 +68,9 @@ static inline int
 __wt_txn_err_chk(WT_SESSION_IMPL *session)
 {
     /* Allow transaction rollback, but nothing else. */
-    if (!F_ISSET(&(session->txn), WT_TXN_ERROR) ||
-      strcmp(session->name, "rollback_transaction") != 0)
+    if (!F_ISSET(&(session->txn), WT_TXN_ERROR) || strcmp(session->name, "WT_CURSOR.close") == 0 ||
+      strcmp(session->name, "WT_CURSOR.reset") == 0 ||
+      strcmp(session->name, "WT_SESSION.rollback_transaction") == 0)
```

Comment:
I have two comments on this. While I'm fine with this if it is the correct change, I think it deserves two changes:
1. The comment is out of date with the exceptions. And I think the exceptions deserve a sentence of explanation about why they're exceptions.
2. Now that we're up to 3 exceptions, maybe add a macro for easy adding/removal of them.

Reply:
I totally agree with your #1, fixed in 3b6108174.

On #2, I'm not sure why a macro makes it easier to add? Can you give me an example?